dns_server.go

```diff
@@ -0,0 +1,106 @@
+package thousandeyes
+
+import (
+	"fmt"
+)
+
+type Server struct {
+	ServerID   int    `json:"serverId,omitempty"`
+	ServerName string `json:"serverName,omitempty"`
+}
+
+type DNSServer struct {
+	Agents                []Agent        `json:"agents,omitempty"`
+	AlertsEnabled         int            `json:"alertsEnabled,omitempty"`
+	AlertRules            []AlertRule    `json:"alertRules,omitempty"`
+	ApiLinks              []ApiLink      `json:"apiLinks,omitempty"`
+	CreatedBy             string         `json:"createdBy,omitempty"`
+	CreatedDate           string         `json:"createdDate,omitempty"`
+	Description           string         `json:"description,omitempty"`
+	Enabled               int            `json:"enabled,omitempty"`
+	Groups                []GroupLabel   `json:"groups,omitempty"`
+	LiveShare             int            `json:"liveShare,omitempty"`
+	ModifiedBy            string         `json:"modifiedBy,omitempty"`
+	ModifiedDate          string         `json:"modifiedDate,omitempty"`
+	SavedEvent            int            `json:"savedEvent,omitempty"`
+	SharedWithAccounts    []AccountGroup `json:"sharedWithAccounts,omitempty"`
+	TestId                int            `json:"testId,omitempty"`
+	TestName              string         `json:"testName,omitempty"`
+	Type                  string         `json:"type,omitempty"`
+	BandwidthMeasurements int            `json:"bandwidthMeasurements,omitempty"`
+	BgpMeasurements       int            `json:"bgpMeasurements,omitempty"`
+	BgpMonitors           []Monitor      `json:"bgpMonitors,omitempty"`
+	DNSServers            []Server       `json:"dnsServers,omitempty"`
+	DNSTransportProtocol  string         `json:"dnsTransportProtocol,omitempty"`
+	Domain                string         `json:"domain,omitempty"`
+	Interval              int            `json:"interval,omitempty"`
+	MtuMeasurements       int            `json:"mtuMeasurements,omitempty"`
+	NetworkMeasurements   int            `json:"networkMeasurements,omitempty"`
+	NumPathTraces         int            `json:"numPathTraces,omitempty"`
+	ProbeMode             string         `json:"probeMode,omitempty"`
+	Protocol              string         `json:"protocol,omitempty"`
+	RecursiveQueries      int            `json:"recursiveQueries,omitempty"`
+}
+
+// AddDNSServer - Add dns server test
+func (t *DNSServer) AddDNSServer(id int) {
```

Comment:
Is this for adding agents or servers?

Reply:
So the type of test is actually called a "dns-server" test

Reply:
https://github.com/william20111/go-thousandeyes/blob/f1ebee8d1f881bd73833455c16f279735899cb67/page_load.go#L52-L55

Cool, just for my sanity this is what we have in the page_load. It adds agents into the test, looks like this is doing the same but with different func name.

Reply:
i think l understand now.  Just pushed a fix.